### PR TITLE
[enigma2.sh] Fix path and detected bootlog after restart enigma2

### PR DESCRIPTION
**Pull request by fairbird**
-Shorten the code, add the missing default path, and correction the recognition of the correct path for the bot logo.
-@sysconfdir@, @prefix@ and @bindir@ doesn't work correctly (broken right path).
-Quotation marks (") also no need it with $?? does not work smoothly with this enigma2.sh script file.

### CODE REVIEW

**Comment by persianpros** [comment]
Just a quick look:

This is wrong, OE manages the path loactions correctly and I didn't see any problem with that.

The part shortens the code is wrong also I think because we should use exteplayer3 only if it exists otherwise use showiframe which is always available.

So no need to check for showiframe as it's always available and we should not use "and/or" because exteplayer3 is optional and when it's not there we should never run bootlogo.mp4 so only one check is required which is exteplayer3 and then run bootlogo.mp4 otherwise always .mvi

**Comment by fairbird** [comment]
1-This is wrong way to focus only with (exteplayer3) and make it the video bootlogo the main bootlog ... not ala user like it !
2-I was test the original code more than 5 times after I get feedback from a lot of friends and does not work correctly ... try to restart GUI you will never see the bootlogo after restart GUI because I as mentioned @sysconfdir@, @Prefix@ and @bindir@ doesn't work correctly (broken right path). And also Quotation marks (").
3-About to check showiframe it is important with this code
> if [ -x "/usr/bin/exteplayer3" -o -x "/usr/bin/showiframe" ]; then
If missing exteplayer3 from image the code will be not working and no need to use long code like this
> if [ -x "@bindir@/exteplayer3" ]; then
		for DIR in "$SEARCHDIRS"
		do
			if [ -d "${DIR}" ] && [ -f "${DIR}/bootlogo.mp4" ]; then
				"@bindir@/exteplayer3" "${DIR}/bootlogo.mp4"
				break
			fi
		done
	else
		if [ -x "@bindir@/showiframe" ]; then
			for DIR in "${SEARCHDIRS}"
			do
				if [ -d "${DIR}" ] && [ -f "${DIR}/bootlogo.mvi" ]; then
					if [ "${SOCFAMILY}" == "hisi" ]; then
						killall -9 showiframe; sleep 1
					fi
					"@bindir@/showiframe" "${DIR}/bootlogo.mvi"
					break
				fi
			done
		fi
fi
4-It is ok for me no problem I can create personal patch to use it with build ... If you did not agree  the pull request you can close it.

**Comment by persianpros** [comment]
I need to test it and see the difference

**Comment by IanSav** [comment]
While I would also like to see the removal of the path obfuscation this would be against the design and coding standards of Enigma2.  If any of these paths were to be changed in the build system then this script could/would become out of step with the rest of Enigma2.

The "/usr/share" directory has been deliberately removed as it encourages clutter of the file system with Enigma2 only related files.  The boot logos are part of the Enigma2 visual / UI system and should, as such, be kept with the other Enigma2 visual / UI system components.

I have previously had issues where "showiframe" has *not* been available.  For OpenVision I was told it will always be available to I relaxed the availability test.  You should never assume that *any* package is installed by default.  I am more than happy to see the availability test returned.

I have no objection to compressing the image and movie search loops into one.  There should be no perceivable performance issues.

File: tools/enigma2.sh.in [hunk 1]
```diff
@@ -63,36 +63,29 @@ fi
 # Enigma main loop.
 while : ; do
 	# Show bootlogo on enigma2 start.
-	SKIN=`sed -En 's|^config\.skin\.primary_skin=(.+)/skin\.xml$|\1|p' "@sysconfdir@/enigma2/settings"`
+	SKIN=`sed -En 's|^config\.skin\.primary_skin=(.+)/skin\.xml$|\1|p' "/etc/enigma2/settings"`
```
Comment (IanSav) [inline tools/enigma2.sh.in:66]: This may cause issues if any of the Enigma2 build system paths are changed.

File: tools/enigma2.sh.in [hunk 2]
```diff
@@ -63,36 +63,29 @@ fi
...
 	if [ -z "${SKIN}" ]; then
-		SKIN=`strings -n 10 "@prefix@/lib/enigma2/python/skin.pyo" | egrep -o -m 1 ".+/skin.xml" | sed 's|/skin.xml.*||'`
+		SKIN=`strings -n 10 "/usr/lib/enigma2/python/skin.pyo" | egrep -o -m 1 ".+/skin.xml" | sed 's|/skin.xml.*||'`
```
Comment (IanSav) [inline tools/enigma2.sh.in:68]: See above.

File: tools/enigma2.sh.in [hunk 3]
```diff
@@ -63,36 +63,29 @@ fi
...
 	fi
 	if [ -n "${SKIN}" ]; then
-		SEARCHDIRS="@sysconfdir@/enigma2/$skin @sysconfdir@/enigma2/skin_common @sysconfdir@/enigma2 @datadir@/enigma2/$skin @datadir@/enigma2/skin_default @datadir@/enigma2"
+		SEARCHDIRS="/etc/enigma2/$skin /etc/enigma2/skin_common /etc/enigma2 /usr/share/enigma2/$skin /usr/share/enigma2/skin_default /usr/share/enigma2 /usr/share"
```
Comment (IanSav) [inline tools/enigma2.sh.in:71]: And again.

File: tools/enigma2.sh.in [hunk 4]
```diff
@@ -63,36 +63,29 @@ fi
...
 	else
-		SEARCHDIRS="@sysconfdir@/enigma2/skin_common @sysconfdir@/enigma2 @datadir@/enigma2/skin_default @datadir@/enigma2"
+		SEARCHDIRS="/etc/enigma2/skin_common /etc/enigma2 /usr/share/enigma2/skin_default /usr/share/enigma2 /usr/share"
```
Comment (IanSav) [inline tools/enigma2.sh.in:73]: See previous comment.

File: tools/enigma2.sh.in [hunk 5]
```diff
@@ -63,36 +63,29 @@ fi
...
 	fi
-	if [ -x "@bindir@/exteplayer3" ]; then
-		for DIR in "$SEARCHDIRS"
+	if [ -x "/usr/bin/exteplayer3" -o -x "/usr/bin/showiframe" ]; then
+		for DIR in $SEARCHDIRS
```
Comment (IanSav) [inline tools/enigma2.sh.in:76]: This is fine.

File: tools/enigma2.sh.in [hunk 6]
```diff
@@ -63,36 +63,29 @@ fi
...
-			if [ -d "${DIR}" ] && [ -f "${DIR}/bootlogo.mp4" ]; then
-				"@bindir@/exteplayer3" "${DIR}/bootlogo.mp4"
+			if [ -d ${DIR} ] && [ -f ${DIR}/bootlogo.mp4 ]; then
+				"/usr/bin/exteplayer3" "${DIR}/bootlogo.mp4"
+				break
+			elif [ -d ${DIR} ] && [ -f ${DIR}/bootlogo.mvi ]; then
+				if [ "${SOCFAMILY}" == "hisi" ]; then
+					killall -9 showiframe; sleep 1
+				fi
+				"/usr/bin/showiframe" "${DIR}/bootlogo.mvi"
```
Comment (IanSav) [inline tools/enigma2.sh.in:85]: This is fine.

File: tools/enigma2.sh.in [hunk 7]
```diff
@@ -63,36 +63,29 @@ fi
...
-			do
-				if [ -d "${DIR}" ] && [ -f "${DIR}/bootlogo.mvi" ]; then
-					if [ "${SOCFAMILY}" == "hisi" ]; then
-						killall -9 showiframe; sleep 1
-					fi
-					"@bindir@/showiframe" "${DIR}/bootlogo.mvi"
-					break
-				fi
-			done
-		fi
```
Comment (IanSav) [inline tools/enigma2.sh.in:95]: It is fine to combine this loop with the one above.

**Review by IanSav — CHANGES_REQUESTED** [review]
While I would also like to see the removal of the path obfuscation this would be against the design and coding standards of Enigma2.  If any of these paths were to be changed in the build system then this script could/would become out of step with the rest of Enigma2.

The "/usr/share" directory has been deliberately removed as it encourages clutter of the file system with Enigma2 only related files.  The boot logos are part of the Enigma2 visual / UI system and should, as such, be kept with the other Enigma2 visual / UI system components.  These boot logo images have been permanently moved to "/usr/share/enigma2/".

I have previously had issues where "showiframe" has *not* been available.  For OpenVision I was told it will always be available to I relaxed the availability test.  You should never assume that *any* package is installed by default.  I am more than happy to see the availability test returned.

I have no objection to compressing the image and movie search loops into one.  There should be no perceivable performance issues.

**Review by IanSav — CHANGES_REQUESTED** [review]
I have marked the changes that are okay as resolved.

I believe that the removal of the build parameters is still problematic.

**Comment by persianpros** [comment]
This is how .sh looks for h9 with today build:

[enigma2.zip](https://github.com/OpenVisionE2/enigma2-openvision/files/6647602/enigma2.zip)

So path is ok, please send a new PR only with the part shortens the code and please allow editing on your PRs so I could do the needed edits instead of having a new PR :+1:

**Pull request closed without merging**